runtime/lua/vim/ui.lua

```diff
@@ -103,4 +103,51 @@ function M.input(opts, on_confirm)
   end
 end
 
+
+--- Prompts the user to pick an option from a list of choices
+---
+---@param msg string Text asked
+---@param opts table
+---     - choices (table|nil)
+---		  List of options, the first letter is mapped as shortcut automatically
+---		  but it can be set by placing `&` before the chosen letter.
+---		  See |confirm()| for additional informations.
+---     - default (number|nil)
+---               Default option selected when pressing `Enter` without pressing any shortcut.
+---@param on_validate function ((idx|nil choice|nil) -> ())
```

Comment:
Let's call the function `on_choice` rather than `on_validate`. Also let's use the official EmmyLua syntax for functions

```suggestion
---@param on_choice fun(idx:number|nil,choice:string|nil)
```